static/js/services/http-wrapper.js

```diff
@@ -18,11 +18,16 @@ var _ = require('underscore');
             return p.url !== url;
           });
         },
-        cancelAll: function() {
+        cancelExceptFor: function(allow) {
+          var stillPending = [];
           _.each(pending, function(p) {
-            p.canceller.resolve();
+            if(_.contains(allow, p.updateTarget)) {
+              stillPending.push(p);
+            } else {
+              p.canceller.reject();
```

Comment:
I don't think this is correct. We're not resolving the request here, we're resolving the request canceller. If you have a look at the `timeout` property of the config parameter of the $http service it says "promise that should abort the request when resolved". So I think you have to `resolve` the timeout promise to cancel the request.